sdks/python/apache_beam/runners/direct/sdf_direct_runner.py

```diff
@@ -464,19 +461,15 @@ def initiate_checkpoint():
       with self._checkpoint_lock:
         if checkpoint_state.checkpointed:
           return
-      checkpoint_state.residual_restriction = tracker.checkpoint()
-      checkpoint_state.checkpointed = object()
+        checkpoint_state.checkpointed = object()
```

Comment:
This is a bug fix, we should be setting checkpoint_state.checkpointed while holding the lock.